src/server/ParsingHttpHandler.ts

```diff
@@ -24,25 +27,33 @@ export interface BaseHttpHandlerArgs {
    * Writes out the response of the operation.
    */
   responseWriter: ResponseWriter;
+  /**
+   * Source handler to send the operation to.
+   */
+  source: OperationHttpHandler;
 }
 
 /**
- * Parses requests and sends the resulting Operation to the abstract `handleOperation` function.
+ * Parses requests and sends the resulting Operation to wrapped source.
  * Errors are caught and handled by the Errorhandler.
- * In case the `handleOperation` function returns a result it will be sent to the ResponseWriter.
+ * In case the source returns a result it will be sent to the ResponseWriter.
  */
-export abstract class BaseHttpHandler extends HttpHandler {
-  protected readonly logger = getLoggerFor(this);
+export class ParsingHttpHandler extends HttpHandler {
+  private readonly logger = getLoggerFor(this);
 
-  protected readonly requestParser: RequestParser;
-  protected readonly errorHandler: ErrorHandler;
-  protected readonly responseWriter: ResponseWriter;
+  private readonly requestParser: RequestParser;
+  private readonly errorHandler: ErrorHandler;
+  private readonly responseWriter: ResponseWriter;
+  private readonly operationMetadataCollector: OperationMetadataCollector;
+  private readonly source: OperationHttpHandler;
```

Comment:
Maybe rename?

Reply:
To what? I just the name `source` here to be similar to how we chain resource stores together (where we also use the `source` parameter for the nested entry).

Reply:
`operationHandler`, I think.